src/components/EditFeatureDrawer/EditReferenceTable/index.tsx

```diff
@@ -0,0 +1,379 @@
+        'This might result in unexpected behaviour');
+    }
+
+    return valueCopy;
+  };
+
+  const onAddClick = () => {
+    const feature = new OlFeature();
+    // We need to set a temporarily internal id on the feature to have it
+    // detectable while it's not being saved and has no id property set.
+    feature.set('internalId', getUid(feature));
+
+    form.setFieldsValue(feature.getProperties());
+
+    setSelectedFeature(feature);
+    setIsModalOpen(true);
+  };
+
+  const onConfirmDelete = (record: EditReferenceDataType) => {
+    const parentFormValues = parentForm.getFieldsValue();
+
+    if (!parentFormValues[propertyName]) {
+      Logger.warn(`Could not find the property ${propertyName} in the form values`);
+      return;
+    }
+
+    const deleteCandidates = _cloneDeep(parentFormValues[propertyName]);
+
+    if (!Array.isArray(deleteCandidates)) {
+      Logger.warn('The delete candidate is not of required type array');
+      return;
+    }
+
+    const deleteIdx = deleteCandidates.findIndex(deleteCandidate => {
+      if (Number.isFinite(deleteCandidate.id) && (deleteCandidate.id === record.id)) {
+        return true;
+      }
+
+      if (deleteCandidate.internalId && (deleteCandidate.internalId === record.internalId)) {
+        return true;
+      }
+
+      return false;
+    });
+
+    if (deleteIdx === -1) {
+      Logger.warn(`Could not find the index (${record.id}) to delete`);
+      return;
+    }
+
+    deleteCandidates.splice(deleteIdx, 1);
+
+    onChange(deleteCandidates);
+  };
+
+  const onDetailsClick = (record: EditReferenceDataType) => {
+    const feature = new OlFeature();
+    feature.setProperties(record);
+
+    form.setFieldsValue(feature.getProperties());
+
+    setSelectedFeature(feature);
+    setIsModalOpen(true);
+  };
+
+  const updateReferences = () => {
+    const formValues = _cloneDeep<EditReferenceDataType>(form.getFieldsValue(true));
+
+    let existingReferences: EditReferenceDataType[] = [];
+    if (Array.isArray(parentForm.getFieldValue(propertyName))) {
+      existingReferences = _cloneDeep(parentForm.getFieldValue(propertyName));
+    }
+
+    const targetIdx = existingReferences.findIndex(existingReference => {
+      if (Number.isFinite(existingReference.id) && (existingReference.id === formValues.id)) {
+        return true;
+      }
+
+      if (existingReference.internalId && (existingReference.internalId === formValues.internalId)) {
+        return true;
+      }
+
+      return false;
+    });
+
+    if (targetIdx > -1) {
+      // Update
+      existingReferences[targetIdx] = formValues;
+    } else {
+      // Add
+      existingReferences.push(formValues);
+    }
+
+    if (form.isFieldsTouched()) {
+      onChange(existingReferences);
+    }
+  };
+
+  const onCancel = () => {
+    closeModal();
+  };
+
+  const onOk = async () => {
+    try {
+      await form.validateFields();
+    } catch (error) {
+      return;
```

Comment:
is the silent return intentional? otherwise we should log something here

Reply:
Added a logger.